### PR TITLE
Andrew/button options

### PR DESCRIPTION
**Pull request by andrewreifman**
I added a new 'ghost' variant to the Button and ButtonLink components. I also added 2 more size options for each component: "x-small" and "icon-small".

CleanShot 2024-12-05 at 15 43 12@2x

### CODE REVIEW

**Review by drewpledger — APPROVED** [review]
LGTM

**Review by migueloller — DISMISSED** [review]
@andrewreifman, this LGTM. I will say, though, I think our buttons need active states (i.e., what they look like while being pressed). Right now it feels weird they don't respond to clicks.

Also, can we add the size variants to the example page in the docs? I was testing things out in the preview deployment but the example doesn't show the new variants so I wasn't able to test those.

**Review by jorgemoya — APPROVED** [review]

**Comment by andrewreifman** [comment]
> @andrewreifman, this LGTM. I will say, though, I think our buttons need active states (i.e., what they look like while being pressed). Right now it feels weird they don't respond to clicks.
> 
> Also, can we add the size variants to the example page in the docs? I was testing things out in the preview deployment but the example doesn't show the new variants so I wasn't able to test those.

I think I've never liked active states because they feel like this glitchy thing you see for 50ms. I feel like me actually clicking a button, plus the action that happens right after is enough affordance to let me know I clicked it. And if there is an extended delay on an action, we have the loading state on the Button.

We've talked about adding more previews within each component's doc, so we could display other props like size, but I could go ahead and add it to the main preview for now.

**Comment by migueloller** [comment]
> > @andrewreifman, this LGTM. I will say, though, I think our buttons need active states (i.e., what they look like while being pressed). Right now it feels weird they don't respond to clicks.
> > Also, can we add the size variants to the example page in the docs? I was testing things out in the preview deployment but the example doesn't show the new variants so I wasn't able to test those.
> 
> I think I've never liked active states because they feel like this glitchy thing you see for 50ms. I feel like me actually clicking a button, plus the action that happens right after is enough affordance to let me know I clicked it. And if there is an extended delay on an action, we have the loading state on the Button.
> 
> We've talked about adding more previews within each component's doc, so we could display other props like size, but I could go ahead and add it to the main preview for now.

The active state is present while the mouse is pressed down, though. Maybe you're a super fast clicker, but not everybody is 😄

**Comment by jorgemoya** [comment]
+1 on updating the examples to play with variants

**Comment by andrewreifman** [comment]
@migueloller For you my friend. It's the same for the Button preview.

CleanShot 2024-12-05 at 16 29 32@2x

**Review by migueloller — DISMISSED** [review]
Nice work @andrewreifman ❤️

File: .github/workflows/dismiss_stale_approvals.yml
Comment (migueloller) [inline .github/workflows/dismiss_stale_approvals.yml:1]: This file shouldn't be in this PR since it's already in `main` 🤔

**Review by migueloller — DISMISSED** [review]

**Review by migueloller — APPROVED** [review]

**Review by jorgemoya — APPROVED** [review]

**Pull request merged**